numpy/core/src/multiarray/compiled_base.c

```diff
@@ -1210,7 +1210,25 @@ arr_unravel_index(PyObject *self, PyObject *args, PyObject *kwds)
     int i, ret_ndim;
     npy_intp ret_dims[NPY_MAXDIMS], ret_strides[NPY_MAXDIMS];
 
-    char *kwlist[] = {"indices", "dims", "order", NULL};
+    char *kwlist[] = {"indices", "shape", "order", NULL};
+
+    /* Continue to support the older "dims" argument in place
+     * of the "shape" argument. Issue an appropriate warning
+     * if "dims" is detected in keywords, then replace it with
+     * the new "shape" argument and continue processing as usual */
+
+    PyObject * dims_item;
+
+    if (kwds) {
+        dims_item = PyDict_GetItemString(kwds, "dims");
+        if (dims_item != NULL) {
+            PyErr_WarnEx(PyExc_RuntimeWarning,
+                         "'shape' argument should be used instead of 'dims'",
+                         1);
```

Comment:
This should be written
```
if (DEPRECATE("the message") < 0) {
    return NULL;
}
```
which correctly deals with warnings being raised as exceptions